src/gtest/test_mempool.cpp

```diff
@@ -136,3 +136,66 @@ TEST(Mempool, TxInputLimit) {
     EXPECT_EQ(state4.GetRejectReason(), "bad-txns-version-too-low");
 }
 
+
+TEST(Mempool, TxExpired) {
+    CTxMemPool pool(::minRelayTxFee);
+    bool missingInputs;
+
+    CMutableTransaction mtx;
+    mtx.fOverwintered = true;
+    mtx.nVersion = 3;
+    mtx.nFormatBranchId = 0xDEADBEEF;
+    // FIXME: Eventually replace with
+    // NetworkUpgradeInfo[Consensus::UPGRADE_OVERWINTER].nBranchId;
+
+    // There is no active chain so expiry height is being compared to chainActive.Height() which is -1
+    // To simulate an expired tx, we must set an expiry height less than -1, that is, set a uint32 value
+    // which when cast to int is less than -1.
+    mtx.nExpiryHeight = 0xFFFFFFFE; // -2 as int
+    CValidationState state1;
+    CTransaction tx1(mtx);
+    EXPECT_FALSE(AcceptToMemoryPool(pool, state1, tx1, false, &missingInputs));
+    EXPECT_EQ(state1.GetRejectReason(), "tx-expired");
+
+    // Check that when 0xFFFFFFFF is set, tx expiry is ignored, and catch the next error about missing inputs.
+    mtx.nExpiryHeight = std::numeric_limits<uint32_t>::max();
+    CValidationState state2;
+    CTransaction tx2(mtx);
+    EXPECT_FALSE(AcceptToMemoryPool(pool, state2, tx2, false, &missingInputs));
+    EXPECT_EQ(state2.GetRejectReason(), "bad-txns-vin-empty");
+
+    // Check a transaction has not expired, and catch the next error about missing inputs.  Set an expiry height > -1.
```

Comment:
999 is >= -1, so this should be an expired transaction...?

Reply:
Do you have a python QA test in your PR?  It would be better to rely on those for actually testing that expiry is working as intended.

Reply:
Haven't finished, will try to push it up today.